base/log_keys.go

```diff
@@ -155,17 +157,28 @@ func ToLogKey(keysStr []string) LogKey {
 		// Strip a single "+" suffix in log keys and warn (for backwards compatibility)
 		if strings.HasSuffix(name, "+") {
 			newName := strings.TrimSuffix(name, "+")
-			Warnf(KeyAll, "Deprecated log key: %q found. Changing to: %q.", name, newName)
+
+			// Need to take a copy of name, as the loop will mutate the value
+			nameCpy := name
```

Comment:
Are you sure this is actually needed since [Go strings are immutable](https://groups.google.com/d/msg/golang-nuts/naMCI9Jt6Qg/4QUebtQz0YAJ)?  I think the later statement `name = newName` shouldn't affect adding a closure function to `deferredLogs`

Reply:
Probably a poorly written comment on my part, as I don't think it's actually mutating name, rather replacing the reference to it. I could add an argument to the closure/anonymous function, but that would require changing the signature of `DeferredLog`.

Assigning a new variable is the other suggested solution in https://golang.org/doc/faq#closures_and_goroutines

>  Even easier is just to create a new variable, using a declaration style that may seem odd but works fine in Go:
> ```go
> for _, v := range values {
>         v := v // create a new 'v'.
>         go func() {
>             fmt.Println(v)
>             done <- true
>         }()
>     }
> ```

Reply:
But I think that's only relevant when there's another goroutine in play.  Here I didn't see one .. although maybe I didn't trace back far  enough.

Reply:
It can occur when goroutines are used, but more generally, when closures are executed after the loop as exited.

Reply:
Here's a way to refactor the code to eliminate the need for copying that value to workaround the closure scope issue.   (I started to write a description but ended up sketching it out in some code)

https://github.com/couchbase/sync_gateway/blob/feature/issue_3510_remove_var_copy/base/log_keys.go#L144-L193

I verified that the test does still pass.

Threw in some function / variable renaming suggestions too.. but don't feel all that strongly about those, so if you like this change feel free to cherry pick the parts you like and skip the renaming changes.